tests/validations/mta/mtaDiagnostic.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { Disposable, mockVscode, testVscode } from "../../mockUtil";
+mockVscode("src/validations/mta/mtaDiagnostic");
+import * as sinon from "sinon";
+import { expect } from "chai";
+import { mta } from "@sap/mta-lib";
+import {
+  clearCurrentDiagnosticCollections,
+  clearDiagnosticCollections,
+  convertMtaIssueCoordinateToEditorCoordinate,
+  diagnosticCollections,
+  getDiagnosticsCollection,
+  mtaIssueToEditorCoordinate,
+} from "../../../src/validations/mta/mtaDiagnostic";
+import { keys } from "lodash";
+
+describe("mtaDiagnostic", () => {
+  const disposables: Disposable[] = [];
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe("diagnosticCollections tests", async () => {
+    beforeEach(() => {
+      clearCurrentDiagnosticCollections();
+    });
+
+    it("getDiagnosticsCollection returns cached DiagnosticCollection", async () => {
+      const diagnosticsName = "diagnosticsName";
+      const diagnosticCollection = {};
+      const createDiagnosticCollectionSpy = sinon
+        .stub(testVscode.languages, "createDiagnosticCollection")
+        .returns(diagnosticCollection);
+
+      // cache is empty
+      expect(keys(diagnosticCollections).length).to.equal(0);
+
+      // first call should create the diagnosticCollection
+      const firstCallDc = getDiagnosticsCollection(
+        diagnosticsName,
+        disposables
+      );
+
+      // make sure the diagnosticCollection was created and not cached
```

Comment:
```suggestion
      // make sure the diagnosticCollection was created and cached
```

Reply:
changed to 
`// make sure the diagnosticCollection was created and not retrieved from cache`